File: factorial/python/factorial-for/factorial.py
#! /usr/bin/env python3

def fact(n):
    if n < 0:
        raise ValueError
    elif n in [0, 1]:
        return 1
    else:
        f = 1
        for i in range(2, n+1):
            f *= i
        return f

for i in range(-1, 11):
    try:
        f = fact(i)
    except ValueError:
        f = "Not defined"
    print("{:2} {:>15}".format(i, f))

File: factorial/ruby/factorial/factorial.rb
#! /usr/bin/env ruby

def factorial(n)
  if n < 0
    raise ArgumentError, "factorial is not defined for negative values"
  elsif n < 2
    1
  else
    f = 1
    n.downto(2) { |i| f *= i }    
    f
  end
end

-1.upto(10) do |i|
  begin
    f = factorial(i).to_s
  rescue ArgumentError
    f = "Not defined"
  end
  puts "%2d % 15s" % [i, f]
end
